Test/pruning/UsesClauses.bpl.pruned.after-doesNotHaveAxioms.expect

```diff
@@ -0,0 +1,33 @@
+const unique four: int;
+
+function ConsumerFunc(x: int) : int;
+
+procedure hasAxioms();
+  requires ProducerFunc(2);
+  requires ProducerConst;
+  ensures ConsumerFunc(4) == 3;
+  ensures four == 4;
+
+
+
+implementation hasAxioms()
+{
+    assert ConsumerFunc(4) == 3;
+    assert four == 4;
+}
```

Comment:
Why is `hasAxioms` printed in the `*-doesNotHaveAxioms` output? The `implementation` is not needed at all. The `procedure` is not used. And since these mention `ProducerConst` and `ProducerFunc`, the `*after-*` file prints as as an invalid Boogie program. Wouldn't it be useful if what's printed here does not include unused (`procedure`? and) `implementation` declarations and is something you can feed back into Boogie as a parsing, resolving program?

Reply:
> Wouldn't it be useful if what's printed here does not include unused (procedure? and) implementation declarations and is something you can feed back into Boogie as a parsing, resolving program?

Yes, but note that pruning works on a split, which may include only part of an implementation and thus prune some things that are required for the implementation.